src/inline.jl

```diff
@@ -47,14 +48,29 @@ function limitstringmime(mime::MIME, x)
     return String(take!(buf))
 end
 
+const ipy_mime_json = [
+    "application/vnd.dataresource+json",
+    "application/vnd.vegalite.v2+json",
+    "application/vnd.vega.v3+json",
+    "application/vnd.vega.v4+json",
+]
+_display_dict(m::MIME, m_str, x) = Dict(m_str=>limitstringmime(m, x))
+# escape JSON string correctly before send_ipython
+for mime in ipy_mime_json
+    @eval begin
+        _display_dict(m::MIME{Symbol($mime)}, m_str, x) = Dict(m_str=>JSON.JSONText(limitstringmime(m, x)))
+        Base.Multimedia.istextmime(::MIME{Symbol($mime)}) = true
```

Comment:
I'm worried that this will conflict with other packages, since we don't "own" this type.

Maybe define `_istextmime(::MIME{Symbol($mime)}) = true` here, along with a fallback definition
```jl
_istextmime(m) = istextmime(m)
```
and then call `_istextmime` rather than `textmime` elsewhere in IJulia.